app/assets/javascripts/controllers/flavor_credentials/flavor_form_controller.js

```diff
@@ -0,0 +1,105 @@
+ManageIQ.angular.app.controller('flavorFormController', ['$scope', '$http', 'flavorId', 'miqService', 'API', function($scope, $http, flavorId, miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.afterGet = false;
+
+    vm.flavorModel = {
+      name: '',
+      ram: '',
+      vcpus: '',
+      disk: '',
+      swap: '',
+      rxtx_factor: '',
+      ems_id: '',
+    };
+
+    vm.model = 'flavorModel';
+    vm.ems_list = [];
+
+    $scope.newRecord = flavorId === 'new';
+
+    ManageIQ.angular.scope = vm;
+
+    $http.get('/flavor/ems_list')
+      .then(getEmsFormDataComplete)
+      .catch(miqService.handleFailure);
+
+    API.get('/api/providers?collection_class=ManageIQ::Providers::EmbeddedAutomationManager')
+      .then(getManagerResource)
+      .catch(miqService.handleFailure);
+  };
+
+  $scope.cancelClicked = function() {
+    miqService.sparkleOn();
+    var message = __('Add of Flavor cancelled by user.');
+    var url = '/flavor/show_list';
+    miqFlashLater({
+      message: message,
+      level: 'warning',
+    });
+    window.location.href = url;
+  };
+
+  $scope.addClicked = function() {
+    miqService.sparkleOn();
+    API.post('/api/flavors/', vm.flavorModel)
+      .then(getBack)
+      .catch(miqService.handleFailure);
+  };
+
+  function setForm() {
+    vm.modelCopy = angular.copy(vm.flavorModel);
+    vm.afterGet = true;
+  }
+
+  var getBack = function(response) {
+    var message = '';
+    var err = false;
+    if (response.hasOwnProperty('results')) {
+      err = ! response.results[0].success;
+      if (! err) {
+        message = sprintf(__('Add of Flavor \"%s\" was successfully initialized.'), vm.flavorModel.name);
+      } else {
+        message = __('Unable to add Flavor ') + vm.flavorModel.name + ' .' + response.results[0].message;
+      }
+    }
+
+    var url = '/flavor/show_list';
+    if (! err) {
+      miqFlashLater({
+        message: message,
+      });
+      window.location.href = url;
+    } else {
+      miqService.miqFlash('error', message);
+      miqService.sparkleOff();
+    }
+  };
+
+  var getManagerResource = function(response) {
+    if (response.resources.length) {
+      vm.flavorModel.manager_resource = {
+        'href': response.resources[0].href,
+      };
+    } else {
+      miqService.miqFlash('error', __('Flavor Provider not found.'));
+    }
+
+    setForm();
+    miqService.sparkleOff();
+  };
+
+  function getEmsFormDataComplete(response) {
```

Comment:
You're changing the model here, but `setForm` can already have been called in `getManagerResource` .. I guess you want to wait for both requests to suceed before calling `setForm` and `sparkleOff`.

(`$q.all` is your friend :))